File: frontend/modules/mediapipe-pose/android/src/main/java/expo/modules/mediapipepose/MediaPipePoseModule.kt
package expo.modules.mediapipepose

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * MediaPipePoseModule — Definicao do modulo Expo para Android.
 *
 * Registra a view nativa (CameraX + MediaPipe) e expoe props/eventos para o JS.
 * O processamento MediaPipe roda inteiramente em threads nativos (nunca no JS thread).
 */
class MediaPipePoseModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("MediaPipePose")

    View(MediaPipePoseView::class) {
      Events(
        "onPoseDetected",
        "onError",
        "onCameraReady"
      )

      Prop("cameraFacing") { view: MediaPipePoseView, facing: String? ->
        view.setCameraFacing(facing ?: "back")
      }

      Prop("isActive") { view: MediaPipePoseView, active: Boolean? ->
        view.setIsActive(active ?: false)
      }

      Prop("modelComplexity") { view: MediaPipePoseView, complexity: Int? ->
        view.setModelComplexity(complexity ?: 0)
      }

      Prop("minDetectionConfidence") { view: MediaPipePoseView, confidence: Double? ->
        view.setMinDetectionConfidence(confidence?.toFloat() ?: 0.6f)
      }

      Prop("minTrackingConfidence") { view: MediaPipePoseView, confidence: Double? ->
        view.setMinTrackingConfidence(confidence?.toFloat() ?: 0.6f)
      }
    }
  }
}

File: frontend/modules/mediapipe-pose/ios/MediaPipePoseModule.swift
import ExpoModulesCore

/**
 * MediaPipePoseModule — Definicao do modulo Expo.
 *
 * Registra a view nativa (camera + MediaPipe) e expoe metodos/eventos para o JS.
 * O processamento MediaPipe roda inteiramente no thread nativo de captura.
 */
public class MediaPipePoseModule: Module {
  public func definition() -> ModuleDefinition {
    Name("MediaPipePose")

    // View nativa com camera + pose detection
    View(MediaPipePoseView.self) {
      Events(
        "onPoseDetected",
        "onError",
        "onCameraReady"
      )

      Prop("cameraFacing") { (view: MediaPipePoseView, facing: String?) in
        view.setCameraFacing(facing ?? "back")
      }

      Prop("isActive") { (view: MediaPipePoseView, active: Bool?) in
        view.setIsActive(active ?? false)
      }

      Prop("modelComplexity") { (view: MediaPipePoseView, complexity: Int?) in
        view.setModelComplexity(complexity ?? 0)
      }

      Prop("minDetectionConfidence") { (view: MediaPipePoseView, confidence: Double?) in
        view.setMinDetectionConfidence(Float(confidence ?? 0.6))
      }

      Prop("minTrackingConfidence") { (view: MediaPipePoseView, confidence: Double?) in
        view.setMinTrackingConfidence(Float(confidence ?? 0.6))
      }
    }
  }
}
